src/crepe/api/SaveManager.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <memory>
+
+#include "../ValueBroker.h"
+
+namespace crepe {
+
+class DB;
+
+class SaveManager {
+public:
+	//! Get a reference to a value and initialize it with a value if it does not yet exist
+	template <typename T>
+	ValueBroker<T> get(const std::string & key, const T & default_value);
+
+	//! Get a reference to a value
+	template <typename T>
+	ValueBroker<T> get(const std::string & key);
+
+	//! Set a value directly
+	template <typename T>
+	void set(const std::string & key, const T & value);
+
+	//! Check if the save file has a value for this \c key
+	bool has(const std::string & key);
+
+private:
+	SaveManager();
+	virtual ~SaveManager() = default;
+
+private:
+	template <typename T>
+	std::string serialize(const T &);
+
+	template <typename T>
+	T deserialize(const std::string &);
+
+public:
+	// singleton
+	static SaveManager & get_instance();
```

Comment:
blijft dit een singleton?

Reply:
Deze klas is momenteel alleen een singleton zodat het voor de gebruiker makkelijker is om de zelfde save file op verschillende plekken in de game te gebruiken, maar misschien moet dit net als de systemen een member van de loop manager worden. Dan zou ik de loop manager wel graag willen hernoemen naar Entrypoint of Main oid.

Reply:
ik vind het niet erg om hem anders te noemen